### ISSUE TITLE
Improve resampling

### ISSUE THREAD

**Issue by gabrieag** [gabrieag/change-detec#2]
Rather than just dropping the particles with the smallest weights, implement stratified optimal resampling according to page 10 of [Fearnhead and Liu (2007)](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.63.8413&rep=rep1&type=pdf).

**Comment by asherbender** [comment]
Worth looking at... the behaviour could be toggled by a key-word argument switch?

**Comment by gabrieag** [comment]
Yes, this is just a reminder for myself. I have a working implementation of this, but the style and variable naming is a bit crude. Once the PR is merged, I'll add this in proper PEP8 style.
